src/vtk_calculator.py

```diff
@@ -44,25 +44,36 @@ def parse_args():
                         help="The mesh (VTK Unstructured Grid) used as input")
     parser.add_argument("--function", "-f", dest="function", required=True,
                         help="""The function to evalutate on the mesh.
-            Syntax is the same as used in the calculator object, coordinates are given as e.g.  'cos(x)+y'.""")
+            Syntax is the same as used in the calculator object, coordinates are given as e.g.  'cos(x)+y'.
+            Alternatively, specify one of the following predefined functions: 'franke' 'eggholder' 'rosenbrock'.""")
     parser.add_argument("--output", "-o", dest="out_meshname", default=None, help="""The output meshname.
             Default is the same as for the input mesh""")
     parser.add_argument("--data", "-d", dest="data", default="MyData", help="""The name of output data.
-            Default is MyData""")
+            Default is \"MyData\".""")
     parser.add_argument("--diffdata", "-diffd", dest="diffdata", help="""The name of difference data.
             Used in diff mode. If not given, output data is used.""")
     parser.add_argument("--log", "-l", dest="logging", default="INFO",
                         choices=["DEBUG", "INFO", "WARNING", "ERROR", "CRITICAL"], help="""Set the log level.
             Default is INFO""")
     parser.add_argument("--directory", "-dir", dest="directory", default=None,
                         help="Directory for output files (optional)")
-    parser.add_argument("--diff", action='store_true', help="Calculate the difference to present data.")
+    parser.add_argument("--diff", action='store_true',
+                        help="Calculate the difference to present data.")
     parser.add_argument("--stats", "-s", action='store_true',
                         help="Store stats of the difference calculation as the separate file inputmesh.stats.json")
     args = parser.parse_args()
     return args
 
 
+preDefFunctions = {
+    "franke": """0.75*exp(-(9*x-2)^2/4-(9*y-2)^2/4)
+                +0.75*exp(-(9*x+1)^2/49-(9*y+1)/10)
+                +0.5*exp(-(9*x-7)^2/4-(9*y-3)^2/4)
+                -0.2*exp(-(9*x-4)^2-(9*y-7))""",
+    "eggholder": "-(y+47)*sin(sqrt(abs(y+x/2+47)))-x*sin(sqrt(abs(x-(y+47))))",
+    "rosenbrock": "(100*(y-x^2)^2+(x-1)^2)+(100*(z-y^2)^2+(y-1)^2)"}
```

Comment:
We probably need to distinguish here between 2D and 3D. Currently, the used Rosenbrock function is 3D, the other two 2D. At least for Franke's function, there is also a 3D variant (e.g. https://link.springer.com/article/10.1007/s10915-017-0431-x).

Reply:
Should we use all functions in 3D? Since we don't know if it is a 2D case it lies in which plane, but pseudo-3D for preCICE?

Reply:
No, I think we should properly distinguish between 2D and 3D. I don't know the ASTE architecture too well, but when calling this function you must know whether you do a 2D or a 3D scenario, right?

Reply:
This part just only calculates the given function on the given mesh.

The user would know the mesh property. The meshes are pseudo-3D even in 2D meshes have all 3 coordinate components. In a 2D case, the mesh can lie on `x-y` `x-z` or `y-z`  planes. There is no way to distinguish them from each other automatically AFAIK. Hence for the 2D cases, there is 3 different combination that is possible, we can implement them as an option and the user can select like `Franke3D` `Franke2D(xy)` `Franke2D(xz)` `Franke2D(yz)`.  The user will have total control and responsibility in this case.

Reply:
I guess this is the way to go. Otherwise the 'detection' algorithm becomes too complicated. Maybe it is also preferable to list all predefined functions when calling `vtk_calculator.py --function --help` or something similar. The default help message might get lengthy otherwise.

Reply:
Can't we simply generate the functions on the fly? After all, they are simple strings.

As @DavidSCN mentioned, something like `--list-functions` would be a simple solution to the readability issue.

Reply:
The help and another argument cannot be taken at same time but I changed somehow it works now 

```bash
./vtk_calculator.py --help
usage: vtk_calculator.py [-h] [--mesh IN_MESHNAME] [--function FUNCTION] [--output OUT_MESHNAME] [--data DATA] [--diffdata DIFFDATA] [--log {DEBUG,INFO,WARNING,ERROR,CRITICAL}]
                         [--directory DIRECTORY] [--diff] [--stats] [--functions]

optional arguments:
  -h, --help            show this help message and exit
  --mesh IN_MESHNAME, -m IN_MESHNAME
                        The mesh (VTK Unstructured Grid) used as input
  --function FUNCTION, -f FUNCTION
                        The function to evalutate on the mesh. Syntax is the same as used in the calculator object, coordinates are given as e.g. 'cos(x)+y'. Default is Egg-Holder
                        function in 3D. Alternatively, you can use predefined function
  --output OUT_MESHNAME, -o OUT_MESHNAME
                        The output meshname. Default is the same as for the input mesh
  --data DATA, -d DATA  The name of output data. Default is "MyData".
  --diffdata DIFFDATA, -diffd DIFFDATA
                        The name of difference data. Used in diff mode. If not given, output data is used.
  --log {DEBUG,INFO,WARNING,ERROR,CRITICAL}, -l {DEBUG,INFO,WARNING,ERROR,CRITICAL}
                        Set the log level. Default is INFO
  --directory DIRECTORY, -dir DIRECTORY
                        Directory for output files (optional)
  --diff                Calculate the difference to present data.
  --stats, -s           Store stats of the difference calculation as the separate file inputmesh.stats.json
  --functions           Prints list of predefined functions.
```

```bash
./vtk_calculator.py --functions
Available predefined functions are:
franke3d
eggholder3d
rosenbrock3d
franke2d(xy)
franke2d(xz)
franke2d(yz)
eggholder2d(xy)
eggholder2d(xz)
eggholder2d(yz)
rosenbrock2d(xy)
rosenbrock2d(xz)
rosenbrock2d(yz)
For more information visit preCICE documentation.
```